sentry-android-core/src/main/java/io/sentry/android/core/ScreenshotEventProcessor.java

```diff
@@ -0,0 +1,106 @@
+package io.sentry.android.core;
+
+import android.app.Activity;
+import android.app.Application;
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.View;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import io.sentry.EventProcessor;
+import io.sentry.SentryEvent;
+import io.sentry.android.core.internal.util.MainThreadChecker;
+import io.sentry.util.Objects;
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.IOException;
+// import java.lang.ref.WeakReference;
+import java.util.HashMap;
+import java.util.Map;
+import org.jetbrains.annotations.NotNull;
+
+public final class ScreenshotEventProcessor
+    implements EventProcessor, Application.ActivityLifecycleCallbacks, Closeable {
+
+  private final @NotNull Application application;
+  private final @NotNull SentryAndroidOptions options;
+  //  TODO: private final @NotNull WeakReference<Activity> currentActivity = new WeakReference();
+  private @Nullable Activity activity;
+
+  public ScreenshotEventProcessor(
+      final @NotNull Application application, final @NotNull SentryAndroidOptions options) {
+    this.application = Objects.requireNonNull(application, "Application is required");
+    this.options = Objects.requireNonNull(options, "SentryAndroidOptions is required");
+
+    if (this.options.isAttachScreenshot()) {
+      application.registerActivityLifecycleCallbacks(this);
+    }
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public @NotNull SentryEvent process(
+      final @NotNull SentryEvent event, @Nullable Map<String, Object> hint) {
+    if (options.isAttachScreenshot()
+        && event.isErrored()
+        && MainThreadChecker.isMainThread(Thread.currentThread())) {
```

Comment:
We could technically do `activity.runOnUiThread(runnable)` that takes the screenshot, but it's again a best effort.